### PR TITLE
3조 송호준: intermediate_source/ddp_tutorial.rst 번역

### PR DESCRIPTION
**Pull request by hojunking**
## 라이선스 동의
_변경해주시는 내용에 BSD 3항 라이선스가 적용됨을 동의해주셔야 합니다._<br />
_더 자세한 내용은 [기여하기 문서](https://github.com/PyTorchKorea/tutorials-kr/blob/master/CONTRIBUTING.md)를 참고해주세요._<br />
_동의하시면 아래 `[ ]`를 `[x]`로 만들어주세요._<br />

- [x] 기여하기 문서를 확인하였으며, 본 PR 내용에 BSD 3항 라이선스가 적용됨에 동의합니다.

## 관련 이슈 번호
_이 Pull Request와 관련있는 이슈 번호를 적어주세요._<br />
_이슈 또는 PR 번호 앞에 #을 붙이시면 제목을 바로 확인하실 수 있습니다. (예. #999 )_

- **이슈 번호**: #(숫자)

## PR 종류
_이 PR에 해당되는 종류 앞의 `[ ]`을 `[x]`로 변경해주세요._<br />
- [ ] 오탈자를 수정하거나 번역을 개선하는 기여
- [x] 번역되지 않은 튜토리얼을 번역하는 기여
- [ ] 공식 튜토리얼 내용을 반영하는 기여
- [ ] 위 종류에 포함되지 않는 기여

## PR 설명
_이 PR로 무엇이 달라지는지 대략적으로 알려주세요._
intermediate_source/ddp_tutorial.rst ddp tutorial 내용 번역

### CODE REVIEW

**Review by hadh93 — COMMENTED** [review]
번역하시느라 고생 많으셨습니다!
- [용어집](https://github.com/PyTorchKorea/tutorials-kr/blob/master/TRANSLATION_GUIDE.md)에 등재된 단어가 있습니다. 
  - Gradient -> 변화도
- 9번 줄의 문장(View and edit ~)은 반드시 편집하라는 지시가 아니라 편집 링크에 대한 안내에 가까운 것 같아서, 적절히 의역하여 _‘이 튜토리얼을 ~에서 조회 및 편집하실 수 있습니다’_ 등으로 수정 가능할 것 같습니다.
- 음차 표기된 '트리거'와 '디바이스'는 한국어로 번역하고 영어 병기하거나, 아예 번역해도 좋을 것 같습니다.

그 이외에는 LGTM입니다 😄

File: intermediate_source/ddp_tutorial.rst [hunk 1]
```diff
@@ -1,38 +1,23 @@
-Getting Started with Distributed Data Parallel
+
+분산 데이터 병렬 처리 시작하기
 =================================================
-**Author**: `Shen Li <https://mrshenli.github.io/>`_
+**저자**: `Shen Li <https://mrshenli.github.io/>`_
 
-**Edited by**: `Joe Zhu <https://github.com/gunandrose4u>`_
+**편집자**: `Joe Zhu <https://github.com/gunandrose4u>`_
```
Comment (hyoyoung) [inline intermediate_source/ddp_tutorial.rst:6]: 저자, 편집자 정보 뒤에 번역자 정보를 넣어둡니다.
다른 문서들을 살펴봐주세요.

File: intermediate_source/ddp_tutorial.rst [hunk 2]
```diff
@@ -1,38 +1,23 @@
...
-combining DDP with model parallel.
+   |edit| 이 튜토리얼을 `github <https://github.com/pytorch/tutorials/blob/main/intermediate_source/ddp_tutorial.rst>`__ 에서 보고 편집하세요.
+
+사전 요구 사항:
+
+-  `PyTorch 분산 개요 <../beginner/dist_overview.html>`__
+-  `DistributedDataParallel API 문서 <https://pytorch.org/docs/master/generated/torch.nn.parallel.DistributedDataParallel.html>`__
+-  `DistributedDataParallel 노트 <https://pytorch.org/docs/master/notes/ddp.html>`__
+
+`DistributedDataParallel <https://pytorch.org/docs/stable/nn.html#module-torch.nn.parallel>`__ (DDP)는 모듈 수준에서 데이터 병렬 처리를 구현하며, 여러 기계에서 실행될 수 있습니다. DDP를 사용하는 애플리케이션(application)은 여러 프로세스를 생성하고 프로세스 당 하나의 DDP 인스턴스를 생성해야 합니다. DDP는 `torch.distributed <https://pytorch.org/tutorials/intermediate/dist_tuto.html>`__ 패키지의 집합적 통신을 사용하여 변화도(gradient)와 버퍼(buffer)를 동기화합니다. 더 구체적으로, DDP는 ``model.parameters()``에 의해 주어진 각 매개변수에 대한 autograd 훅(hook)을 등록하고, 해당 변화도(gradient)가 역전파(backward)에서 계산될 때 훅(hook)이 발동됩니다. 그 후 DDP는 그 신호를 사용하여 프로세스 간 변화도(gradient) 동기화를 트리거(trigger)합니다. 자세한 내용은 `DDP 설계 노트 <https://pytorch.org/docs/master/notes/ddp.html>`__ 를 참조하십시오.
```
Comment (hyoyoung) [inline intermediate_source/ddp_tutorial.rst:17]: 1. 버퍼같은 경우는 흔히 쓰이는 표현이니 영어 병기가 필요 없을 듯 합니다.
2. 한번 영어를 병기한 용어가 다시 쓰이는 경우, 두번째부터는 영어 병기를 안하는게 보기에 더 편리합니다

File: intermediate_source/ddp_tutorial.rst [hunk 3]
```diff
@@ -1,38 +1,23 @@
-Getting Started with Distributed Data Parallel
+
```
Comment (hyoyoung) [inline intermediate_source/ddp_tutorial.rst:1]: 원문에 없는 공백은 추가하지 않습니다

**Review by hyoyoung — CHANGES_REQUESTED** [review]
전반적으로 번역이 잘 되어있습니다.
기존에 번역된 다른 문서도 참고해보시면 더 좋을 것 같습니다.

몇가지 코멘트를 남기니 확인하시고 차후 번역에 참고 부탁드립니다.

File: intermediate_source/ddp_tutorial.rst (hunk 1 above)
Comment (hojunking) [reply intermediate_source/ddp_tutorial.rst:5]: 저자, 편집자 둘 다 있어서, 번역자 정보를 두 개 넣었습니다. 혹시 하나만 넣는 게 맞을까요?

**Review by hojunking — COMMENTED** [review]
말씀해주신 피드백을 반영했습니다. author, editor 이후 번역자 정보를 추가했고, buffer에 대한 임차 표기는 지우고, 한 번 임차표기한 단어들에 대해서는 이후 임차 표기 없이 번역했습니다.
꼼꼼한 피드백 감사드립니다. 저도 좀 더 꼼꼼하게 번역하도록 노력하겠습니다!

File: intermediate_source/ddp_tutorial.rst (hunk 3 above)
Comment (hojunking) [reply intermediate_source/ddp_tutorial.rst]: 공백 제거했습니다. 다음부턴 확인하겠습니다!
